### PR TITLE
added swipe left/right to close/open navbar

### PR DESCRIPTION
**Pull request by ezwang**
decided not to use library because too many extra features
tested on chrome on android, should work for modern mobile browsers

### CODE REVIEW

File: intranet/static/js/common.nav.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        var nowY = e.originalEvent.touches[0].clientY;
+        if (Math.abs(nowY - initY) > 30) {
+            listening = false;
+            return;
+        }
+        var diffX = nowX - initX;
+        if (Math.abs(diffX) > 30) {
+            var nav = $(".main > .nav").eq(0);
+            var g = $(".nav-g");
+            var shown = nav.css('left').split(/[^\-\d]+/)[0] == 0;
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l25)

File: intranet/static/js/common.nav.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+$(function() {
+    var initX = null, initY = null, listening = false;
+    $(window).on("touchstart", function(e) {
+        initX = e.originalEvent.touches[0].clientX;
+        initY = e.originalEvent.touches[0].clientY;
+        listening = true;
+    });
+    $(window).on("touchend", function(e) {
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ['e' is defined but never used.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l8)

File: intranet/static/js/common.nav.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        }
+        var diffX = nowX - initX;
+        if (Math.abs(diffX) > 30) {
+            var nav = $(".main > .nav").eq(0);
+            var g = $(".nav-g");
+            var shown = nav.css('left').split(/[^\-\d]+/)[0] == 0;
+            if (diffX > 0 && !shown) {
+                nav.animate({ left: "0px" }, 200);
+                g.addClass("close-l").fadeIn(200);
+                $("body").addClass("disable-scroll").addClass("mobile-nav-show");
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l29)

File: intranet/static/js/common.nav.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    var initX = null, initY = null, listening = false;
+    $(window).on("touchstart", function(e) {
+        initX = e.originalEvent.touches[0].clientX;
+        initY = e.originalEvent.touches[0].clientY;
+        listening = true;
+    });
+    $(window).on("touchend", function(e) {
+        listening = false;
+    });
+    $(window).on("touchmove", function(e) {
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: [Function 'anonymous' has a complexity of 6.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l11)

File: intranet/static/js/common.nav.js (hunk 2 above)
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l8)

File: intranet/static/js/common.nav.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+            var shown = nav.css('left').split(/[^\-\d]+/)[0] == 0;
+            if (diffX > 0 && !shown) {
+                nav.animate({ left: "0px" }, 200);
+                g.addClass("close-l").fadeIn(200);
+                $("body").addClass("disable-scroll").addClass("mobile-nav-show");
+            }
+            else if (shown) {
+                nav.animate({ left: "-202px" }, 200);
+                g.removeClass("close-l").fadeOut(200);
+                $("body").removeClass("disable-scroll").removeClass("mobile-nav-show");
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l34)

File: intranet/static/js/common.nav.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+$(function() {
```
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l1)

File: intranet/static/js/common.nav.js (hunk 4 above)
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: ["$" is not defined.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l11)

File: intranet/static/js/common.nav.js (hunk 1 above)
Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: [Strings must use doublequote.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l25)

Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://www.codacy.com/app/tjcsl/ion/file/2286322831/issues/source?bid=3176160&fileBranchId=3189731#l25)

**Comment by jwoglom** [comment]
Merge for test

**Comment by jwoglom** [comment]
@ezwang Suggestion: on the eighth period signup page, if you have the activity info pane showing (with the back arrow on the left), allow a swipe L-R to close it before opening the nav.

**Comment by jwoglom** [comment]
Also, on the same page, a swipe from bottom to top to close the block selector, and top to bottom to open it.

File: intranet/static/js/common.nav.js (hunk 6 above)
Comment (fwilson42) [inline intranet/static/js/common.nav.js]: @pefoley2 you're lying

Comment (pefoley2) [inline intranet/static/js/common.nav.js]: No, I (codacy) is letting you know that you're depending on jquery being included before this file is included.

Comment (jwoglom) [inline intranet/static/js/common.nav.js]: This is still annoying and should be fixed, with I would imagine some
config change in Codacy.

Comment (pefoley2) [inline intranet/static/js/common.nav.js]: Sure, can you add the integration using @tjhsst-ion though?
I'll work on fixing codacy.

**Pull request merged**